examples/pxScene2d/src/pxHttpCache.cpp

```diff
@@ -0,0 +1,406 @@
+  if (isValid())
+  {
+    string cacheControl = mHeaderMap["Cache-Control"].cString();
+    if (string::npos != cacheControl.find("no-store"))
+    {
+      return false;
+    }
+    return true;
+  }
+  return false;
+}
+
+void rtHttpCacheData::setAttributes(const char* rawAttributes)
+{
+  mHeaderMetaData.init(rawAttributes,strlen(rawAttributes));
+  populateHeaderMap();
+  setExpirationDate();
+}
+
+rtError rtHttpCacheData::attributes(map<rtString, rtString>& cacheAttributes)
+{
+  cacheAttributes = mHeaderMap;
+  return RT_OK;
+}
+
+rtData& rtHttpCacheData::getHeaderData()
+{
+  return mHeaderMetaData;
+}
+
+rtData& rtHttpCacheData::getContentsData()
+{
+  return mData;
+}
+
+rtError rtHttpCacheData::data(rtData& data)
+{
+  if (NULL == fp)
+    return RT_ERROR;
+
+  if (mHeaderMap.end() != mHeaderMap.find("ETag"))
+  {
+    bool isUpdated = isDataUpdatedInServer();
+    if (isUpdated)
+    {
+      populateHeaderMap();
+      setExpirationDate();
+      data.init(mData.data(),mData.length());
+      mUpdated = true;
+      fclose(fp);
+      return RT_OK;
+    }
+  }
+
+  char *contentsData = NULL;
+  char buffer[100];
+  int bytesCount = 0;
+  int totalBytes = 0;
+  while (!feof(fp))
+  {
+    bytesCount = fread(buffer,1,100,fp);
+    if (NULL == contentsData)
+      contentsData = (char *)malloc(bytesCount);
+    else
+      contentsData = (char *)realloc(contentsData,totalBytes+bytesCount);
+    if (NULL == contentsData)
+    {
+      fclose(fp);
+      return RT_ERROR;
+    }
+    memcpy(contentsData+totalBytes,buffer,bytesCount);
+    totalBytes += bytesCount;
+    memset(buffer,0,100);
+  }
+  fclose(fp);
+  if (NULL != contentsData)
+  {
+    mData.init(contentsData,totalBytes);
+    free(contentsData);
+    contentsData = NULL;
+    data.init(mData.data(),mData.length());
+  }
+  return RT_OK;
+}
+
+void rtHttpCacheData::setData(const rtData& cacheData)
+{
+  mData.init(cacheData.data(),cacheData.length());
+}
+
+rtError rtHttpCacheData::url(rtString& url)
+{
+  url = mUrl;
+  return RT_OK;
+}
+
+rtError rtHttpCacheData::etag(rtString& tag) //returns the etag (if available)
+{
+  if (mHeaderMap.end() != mHeaderMap.find("Etag"))
+  {
+    tag = mHeaderMap["Etag"];   	
+    return RT_OK;
+  }
+  return RT_ERROR;
+}
+
+bool rtHttpCacheData::isUpdated()
+{
+  return mUpdated;
+}
+
+void rtHttpCacheData::setFilePointer(FILE* openedDescriptor)
+{
+  fp = openedDescriptor;
+}
+
+void rtHttpCacheData::setProxy(rtString& proxyServer)
+{
+  mProxyServer = proxyServer;
+}
+
+void rtHttpCacheData::setExpirationDate()
+{
+  string expirationDate = "";
+  bool foundMaxAge = false;
+  if (mHeaderMap.end() != mHeaderMap.find("Cache-Control"))
+  {
+    string cacheControl = mHeaderMap["Cache-Control"].cString();
+    int pos = cacheControl.find("max-age");
+    if (string::npos != pos)
+    {
+      foundMaxAge = true;
+      string maxAge = cacheControl.substr(pos+8);
+      long int maxAgeInt = stol(maxAge);
```

Comment:
Change from stol to the below code to support all platforms:

long int maxAgeInt = 0;
stringstream stream(maxAge);
stream >> maxAgeInt;